### PR TITLE
tests:  Update to cyclonedx-core-java-9.0.2 for test runners

### PR DESCRIPTION
**Pull request by Nicolas-Peiffer**
See also: https://github.com/CycloneDX/specification/pull/477

~~Please merge this PR after merging https://github.com/CycloneDX/specification/pull/479.~~
PR is atomic.

~~Tests are testing the 1.6 JSON and XML files.~~
this will come in an extra PR

pr-update-java-lib_20240612_122316

### CODE REVIEW

File: schema/xmlcatalog.xml
Comment (jkowalleck) [inline schema/xmlcatalog.xml:1]: this file seams to be slipped in by accident. it is irrelevant for the java upgrade.
please remove it

File: tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java
Comment (jkowalleck) [inline tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java:32]: this file seams to be slipped in by accident. it is irrelevant for the java upgrade.
please remove it and create an extra PR for it

File: tools/src/test/java/org/cyclonedx/schema/XmlCatalogVerificationTest.java
Comment (jkowalleck) [inline tools/src/test/java/org/cyclonedx/schema/XmlCatalogVerificationTest.java:1]: this file seams to be slipped in by accident. it is irrelevant for the java upgrade.
please remove it

File: tools/src/test/java/org/cyclonedx/schema/XmlSchemaVerificationTest.java
Comment (jkowalleck) [inline tools/src/test/java/org/cyclonedx/schema/XmlSchemaVerificationTest.java:56]: this file seams to be slipped in by accident. it is irrelevant for the java upgrade.
please remove it and create an extra PR for it

File: tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java
Comment (jkowalleck) [inline tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java:46]: this file seams to be slipped in by accident. it is irrelevant for the java upgrade.
please remove it and create an extra PR for it

**Review by jkowalleck — CHANGES_REQUESTED** [review]

File: schema/xmlcatalog.xml
Comment (Nicolas-Peiffer) [reply schema/xmlcatalog.xml:1]: If you first merge https://github.com/CycloneDX/specification/pull/479 before merging https://github.com/CycloneDX/specification/pull/480/files, this would not matter.

Nevertheless, if you feel it is better to remove it, I will remove the 2 files:

* `schema/xmlcatalog.xml`
* `tools/src/test/java/org/cyclonedx/schema/XmlCatalogVerificationTest.java`

And I will modify the `pom.xml` not to include the custom `testResource` paths.

Would this be satifying ?

Comment (jkowalleck) [reply schema/xmlcatalog.xml:1]: > If you first merge https://github.com/CycloneDX/specification/pull/479 before merging https://github.com/CycloneDX/specification/pull/480/files, this would not matter.

this is not intended. pull requests should be atomic.

File: tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java
Comment (Nicolas-Peiffer) [reply tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java:46]: In this case, I think you are wrong: upgrading from [`cyclonedx-core-java-8.0.3`](https://github.com/CycloneDX/cyclonedx-core-java/releases/tag/cyclonedx-core-java-8.0.3) to [`cyclonedx-core-java-9.0.2`](https://github.com/CycloneDX/cyclonedx-core-java/releases/tag/cyclonedx-core-java-9.0.2) has an impact on the class `CycloneDxSchema`: some of its content has been moved to a new file.

The `public enum Version` has moved from file `src/main/java/org/cyclonedx/CycloneDxSchema.java` in v8.0.3 to file `src/main/java/org/cyclonedx/Version.java` in v9.0.2.

For **cyclonedx-core-java-9.0.2**: there is a new file `Version.java`: 
https://github.com/CycloneDX/cyclonedx-core-java/blob/cyclonedx-core-java-9.0.2/src/main/java/org/cyclonedx/Version.java

If you only modify the `pom.xml` to upgrade from 8.0.3 to 9.0.2, then the Java tests will fail, as they need the content from the new file `src/main/java/org/cyclonedx/Version.java` that you get from `import org.cyclonedx.Version;`.

If you want me to do an extra PR for this, please elaborate because I don't understand how to do this.

File: tools/src/test/java/org/cyclonedx/schema/XmlSchemaVerificationTest.java
Comment (Nicolas-Peiffer) [reply tools/src/test/java/org/cyclonedx/schema/XmlSchemaVerificationTest.java:56]: In this case, I think you are wrong: upgrading from [`cyclonedx-core-java-8.0.3`](https://github.com/CycloneDX/cyclonedx-core-java/releases/tag/cyclonedx-core-java-8.0.3) to [`cyclonedx-core-java-9.0.2`](https://github.com/CycloneDX/cyclonedx-core-java/releases/tag/cyclonedx-core-java-9.0.2) has an impact on the class `CycloneDxSchema`: some of its content has been moved to a new file.

The `public enum Version` has moved from file `src/main/java/org/cyclonedx/CycloneDxSchema.java` in v8.0.3 to file `src/main/java/org/cyclonedx/Version.java` in v9.0.2.

For **cyclonedx-core-java-9.0.2**: there is a new file `Version.java`: 
https://github.com/CycloneDX/cyclonedx-core-java/blob/cyclonedx-core-java-9.0.2/src/main/java/org/cyclonedx/Version.java

If you only modify the `pom.xml` to upgrade from 8.0.3 to 9.0.2, then the Java tests will fail, as they need the content from the new file `src/main/java/org/cyclonedx/Version.java` that you get from `import org.cyclonedx.Version;`.

If you want me to do an extra PR for this, please elaborate because I don't understand how to do this.

File: tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java
Comment (Nicolas-Peiffer) [reply tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java]: For this particular file, I can create a third PR _[FEATURE] Adding tests for CycloneDX 1.6_.

This PR will only add this line `files.addAll(getResources("1.6/"));`.

File: tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java
Comment (Nicolas-Peiffer) [reply tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java:46]: Okay sorry I misunderstood, I though you were talking about the whole file.

But you understand that the 2 are intricate: if I do a third PR for 1.6 before you propagate the `cyclonedx-core-java` upgrade, this means I need to do extra work.

So I will remove everything related to 1.6, and I will wait for this PR about the java lib update to be merged before doing a new PR for adding 1.6 tests.

File: schema/xmlcatalog.xml
Comment (Nicolas-Peiffer) [reply schema/xmlcatalog.xml:1]: done

File: tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java
Comment (Nicolas-Peiffer) [reply tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java]: I removed `files.addAll(getResources("1.6/"));.` from file `tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java`.

But I will wait for #480  to be merged before adding back 1.6 tests.

**Review by jkowalleck — APPROVED** [review]

File: tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java
Comment (Nicolas-Peiffer) [reply tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java]: Agreed :smiley: . However, I will wait for the PR #480 to be merged, because it will be easier for me to do the PR dedicated for 1.6 after the java lib update is merged.
Otherwise I need to revert to many commits on my end.

Comment (jkowalleck) [reply tools/src/test/java/org/cyclonedx/schema/BaseSchemaVerificationTest.java]: understood. will merge this PR soon.

**Pull request merged**
